valhalla/sif/dynamiccost.h

```diff
@@ -346,21 +346,25 @@ class DynamicCost {
         // Ids do not match the path for this restriction.
         bool match = true;
         const EdgeLabel* next_pred = first_pred;
+        // Remember the edge_ids in restriction for later reset
         std::vector<baldr::GraphId> edge_ids_in_complex_restriction;
         edge_ids_in_complex_restriction.reserve(10);
-        if (cr->via_count() > 0) {
-          // The via list starts immediately after the structure
-          baldr::GraphId* via = reinterpret_cast<baldr::GraphId*>(cr + 1);
-          for (uint32_t i = 0; i < cr->via_count(); i++, via++) {
-            if (via->value != next_pred->edgeid().value) {
-              match = false;
-              break;
-            }
+
+        cr->WalkVias([&match, &next_pred, next_predecessor,
```

Comment:
doesnt look like `next_predecessor` is used and also super confusingly named :smile:

Reply:
It's used on line 362, no change from master in use or naming :)

Reply:
oh man that was hard to see, sorry about that, i looked at it about 5 times...